instrumentation/servlet/servlet-common/library/src/main/java/io/opentelemetry/instrumentation/servlet/ServletAccessor.java

```diff
@@ -52,7 +52,8 @@
 
   boolean isRequestAsyncStarted(REQUEST request);
 
-  void addRequestAsyncListener(REQUEST request, ServletAsyncListener<RESPONSE> listener);
+  void addRequestAsyncListener(
+      REQUEST request, ServletAsyncListener<RESPONSE> listener, Object response);
```

Comment:
Since it's taken from a request attribute, that would require doing a `(RESPONSE) response` unchecked cast in the tracer, which is not really nice and shows a warning in IntelliJ unless a `//noinspection` is added. Since the accessor implementations work with specific classes, they can do the `instanceof` check properly.